### ISSUE TITLE
How to pop up a notification?

### ISSUE THREAD

**Issue by Northern-Lights** [gotk3/gotk3#310]
I'm trying to use glib to pop up a simple notification.  I have the following code to initialize a `gtk.Application` to do so:

```go
package main

import (
	"fmt"
	"os"
	"time"

	"github.com/gotk3/gotk3/glib"
	"github.com/gotk3/gotk3/gtk"
)

const AppID = "com.example.app"

func main() {
	fmt.Println("Creating app")
	app, err := gtk.ApplicationNew(AppID, glib.APPLICATION_FLAGS_NONE)
	if err != nil {
		panic(err)
	}
	fmt.Println("Connecting activate")
	_, err = app.Connect("activate", func() {
		fmt.Println("Activating")
	})
	if err != nil {
		panic(err)
	}
	fmt.Println("Running app")
	app.Run(os.Args)

	_, err = glib.IdleAdd(func() {
		notif := glib.NotificationNew("Notification Title")
		notif.SetBody("BODY")
		notif.SetTitle("TITLE")
		app.SendNotification("notifID", notif)
	})
	if err != nil {
		panic(err)
	}

	fmt.Println("Sleeping")
	time.Sleep(5 * time.Second)
}
```
There are no errors, but no notification ever pops up.  Any help is appreciated.

**Comment by Bios-Marcel** [comment]
You aren't supposed to do that stuff after app.Run afaik. Put it into the activate signal handler instead.

This works:

```go
package main

import (
	"os"

	"github.com/gotk3/gotk3/glib"
	"github.com/gotk3/gotk3/gtk"
)

const appId = "com.example.app"

func main() {
	app, _ := gtk.ApplicationNew(appId, glib.APPLICATION_FLAGS_NONE)

	app.Connect("activate", func() {
		notif := glib.NotificationNew("Notification")
		notif.SetBody("My text!")
		app.SendNotification(appId, notif)
	})

	app.Run(os.Args)
}
```
It'd be a different story if you were manually invoking `gtk.Init` and `gtk.Main` I think, but don't quote me here ;)

**Comment by Northern-Lights** [comment]
There must be something different with my environment, since your example still produces no notification on my system.  I'll have to dig a little bit deeper, but thanks!

**Comment by Bios-Marcel** [comment]
Oh that is interesting. Do other applications give you notifications though? I assume you are on Linux?

**Comment by Northern-Lights** [comment]
Yes, to both. I've also written a simple app using libnotify in C, and it successfully sends notifications, so there must be something going on with glib that I don't know about.

**Comment by Bios-Marcel** [comment]
Hm, weird. Would be cool if you could post the solution in case you solve the problem.

**Comment by ddkv** [comment]
As I understand, Linux GUI desktops do support only libnotify subsystem at the moment to pop up messages.
So, I implement this in my fork of GOTK3, just adding package libnotify.
You can find how this work in application written by me:  [https://github.com/d2r2/go-rsync](url)

**Comment by grandchild** [comment]
Why is this a milestone component for V0.6.0? I don't get what the action needed here is. The changes from @ddkv are a bit impenetrable, because there's 127 commits in d2r2/gotk3 (90% of them Readme updates), but they're also 866 behind, so I don't know if those changes are even rebase-able anymore.

I'm asking because 0.6.0 is long overdue and the fact that official gotk3 doesn't compile anymore with go1.16 breaks basically all projects consuming gotk3 at the moment.
